dev/NumberBox/InteractionTests/NumberBoxTests.cs

```diff
@@ -0,0 +1,335 @@
+                Wait.ForIdle();
+                Verify.AreEqual(0, numBox.Value);
+
+                Log.Comment("Verify that typing '123' in the NumberBox changes the value to 100");
+                EnterText(numBox, "123");
+                Verify.AreEqual(100, numBox.Value);
+
+                Log.Comment("Changing Max to 90; verify value also changes to 90");
+                EnterText(FindElement.ByName<RangeValueSpinner>("MaxNumberBox"), "90");
+                Verify.AreEqual(90, numBox.Value);
+            }
+        }
+
+        [TestMethod]
+        public void BasicCalculationTest()
+        {
+            using (var setup = new TestSetupHelper("NumberBox Tests"))
+            {
+                RangeValueSpinner numBox = FindElement.ByName<RangeValueSpinner>("TestNumberBox");
+
+                Log.Comment("Verify that calculations don't work if AcceptsCalculations is false");
+                EnterText(numBox, "5 + 3");
+                Verify.AreEqual(0, numBox.Value);
+
+                Check("CalculationCheckBox");
+
+                EnterText(numBox, "5 + 3");
+                Verify.AreEqual(8, numBox.Value);
+
+                EnterText(numBox, "9-3*2");
+                Verify.AreEqual(3, numBox.Value);
+
+                EnterText(numBox, "10 /( 2 + 3 )");
+                Verify.AreEqual(2, numBox.Value);
+
+                EnterText(numBox, "2 3 ^");
+                Verify.AreEqual(8, numBox.Value);
+
+                Log.Comment("Verify that diving by zero doesn't work");
+                EnterText(numBox, "1 / 0");
+                Verify.AreEqual(8, numBox.Value);
+            }
+        }
+
+        [TestMethod]
+        public void BasicKeyboardTest()
+        {
+            using (var setup = new TestSetupHelper("NumberBox Tests"))
+            {
+                RangeValueSpinner numBox = FindElement.ByName<RangeValueSpinner>("TestNumberBox");
+
+                Log.Comment("Verify that loss of focus validates textbox contents");
+                EnterText(numBox, "8", false);
+                KeyboardHelper.PressKey(Key.Tab);
+                Wait.ForIdle();
+                Verify.AreEqual(8, numBox.Value);
+
+                Log.Comment("Verify that pressing escape cancels entered text");
+                EnterText(numBox, "3", false);
+                KeyboardHelper.PressKey(Key.Escape);
+                Wait.ForIdle();
+                Verify.AreEqual(8, numBox.Value);
+
+                Log.Comment("Verify that pressing up arrow increases the value");
+                KeyboardHelper.PressKey(Key.Up);
+                Wait.ForIdle();
+                Verify.AreEqual(9, numBox.Value);
+
+                Log.Comment("Verify that pressing down arrow decreases the value");
+                KeyboardHelper.PressKey(Key.Down);
+                Wait.ForIdle();
+                Verify.AreEqual(8, numBox.Value);
+            }
+        }
+
+        [TestMethod]
+        public void GamepadTest()
```

Comment:
>GamepadTest [](start = 20, length = 11)

Can you adjust the value with the up and down buttons using gamepad?

Reply:
I don't believe so because the up and down buttons change focus, don't they?

Reply:
Some controls use a button to put the control in a mode which traps gamepad navigation, for example in list view you can press A(I think) to enter the list view and move within it using the D-pad, otherwise it is skipped over using the d-pad. 

---
In reply to: [341299486](https://github.com/microsoft/microsoft-ui-xaml/pull/1507#discussion_r341299486) [](ancestors = 341299486)

Reply:
I'll add it to the spec questions -- I'm not sure what's wanted in this case.